In [1]:
import pandas as pd
df = pd.read_csv("crawling_data/review_only_100_data/table3_essence_ingredients_productname_normalized copy.csv")

In [2]:
df.head()

product_id   category  brand  \
0       E001  에센스/세럼/앰플    아누아   
1       E002  에센스/세럼/앰플    차앤박   
2       E003  에센스/세럼/앰플    메디힐   
3       E004  에센스/세럼/앰플   메디큐브   
4       E005  에센스/세럼/앰플  이니스프리   

                                        product_name  \
0  [2025 어워즈] 아누아 피디알엔 히알루론산 캡슐 100 세럼 30mL (+리필3...   
1    [1월올영픽/진정탄력] 차앤박 더마앤서 액티브 부스트 PDRN 앰플 30ml 더블기획   
2       [단독/1+1] 메디힐 마데카소사이드 흔적 리페어 세럼 40+40ml 더블 기획   
3  [2025 어워즈/미백천재앰플] 메디큐브 PDRN 핑크앰플 30ml 어워즈 기획 (...   
4            [1월올영픽/키링증정] 이니스프리 레티놀 시카 앰플 30ml 더블 기획   

                                         ingredients  product_rating  \
0  [세럼] 하이드롤라이즈드하이알루로닉애씨드(29,049ppm), 나이아신아마이드, 하...             4.8   
1  정제수, 글리세린, 다이프로필렌글라이콜, 피이지/피피지/폴리부틸렌글라이콜-8/5/3...             4.9   
2  정제수, 글리세린, 부틸렌글라이콜, 프로판다이올, 1,2-헥산다이올, 나이아신아마이...             4.7   
3  PDRN 앰플 정제수,글리세린,다이프로필렌글라이콜,아이소프로필미리스테이트,글리세레스...             4.8   
4  정제수, 글리세린, 부틸렌글라이콜, 나이아신아마이드, 다이부틸아디페이트, 카프릴릭/...             4.8   

                canonical_name  
0     아누아 피디알엔 히알루론산 캡슐 100 세럼  
1  차앤박 더마앤서 액티브 부스트 PDRN 앰플 더블  
2     메디힐 마데카소사이드 흔적 리페어 세럼 40  
3               메디큐브 PDRN 핑크앰플  
4           이니스프리 레티놀 시카 앰플 더블

In [14]:
# canonical_name이 중복된 행들의 product_name을 전부 다 보이게 출력

import pandas as pd

# pandas 출력 옵션 설정 - 모든 행과 열을 다 보이게
pd.set_option('display.max_rows', None)  # 모든 행 표시
pd.set_option('display.max_columns', None)  # 모든 열 표시
pd.set_option('display.max_colwidth', None)  # 열 내용 전체 표시
pd.set_option('display.width', None)  # 출력 너비 제한 없음

# canonical_name이 중복된 것들만 필터링
duplicates = df[df.duplicated(subset=['canonical_name'], keep=False)]

# canonical_name으로 정렬해서 보기 쉽게
duplicates_sorted = duplicates.sort_values('canonical_name')

# 결과 출력
print(f"중복된 canonical_name 개수: {duplicates_sorted['canonical_name'].nunique()}")
print(f"중복된 총 행 개수: {len(duplicates_sorted)}")
print("\n중복된 행들:")
print(duplicates_sorted)

# canonical_name별 그룹으로 product_name 비교
print("\n\n=== canonical_name별 product_name 비교 ===")
for name, group in duplicates_sorted.groupby('canonical_name'):
    print(f"\n[{name}] - {len(group)}개 행")
    print(f"Product IDs: {group['product_id'].tolist()}")
    print("Product Names:")
    for idx, row in group.iterrows():
        print(f"  - {row['product_id']}: {row['product_name']}")
    print("-" * 100)

중복된 canonical_name 개수: 43
중복된 총 행 개수: 94

중복된 행들:
    product_id   category  brand  \
183       E184  에센스/세럼/앰플     VT   
12        E013  에센스/세럼/앰플     VT   
88        E089  에센스/세럼/앰플     VT   
31        E032  에센스/세럼/앰플     VT   
247       E248  에센스/세럼/앰플     구달   
28        E029  에센스/세럼/앰플     구달   
199       E200  에센스/세럼/앰플   넘버즈인   
49        E050  에센스/세럼/앰플   넘버즈인   
30        E031  에센스/세럼/앰플   넘버즈인   
61        E062  에센스/세럼/앰플   넘버즈인   
76        E077  에센스/세럼/앰플    닥터지   
47        E048  에센스/세럼/앰플    닥터지   
269       E270  에센스/세럼/앰플    이지듀   
69        E070  에센스/세럼/앰플    이지듀   
252       E253  에센스/세럼/앰플  더페이스샵   
213       E214  에센스/세럼/앰플  더페이스샵   
266       E267  에센스/세럼/앰플   디마르3   
59        E060  에센스/세럼/앰플   디마르3   
78        E079  에센스/세럼/앰플  디오디너리   
56        E057  에센스/세럼/앰플  디오디너리   
272       E273  에센스/세럼/앰플   마녀공장   
275       E276  에센스/세럼/앰플   마녀공장   
192       E193  에센스/세럼/앰플  메이크프렘   
287       E288  에센스/세럼/앰플  메이크프렘   
160       E161  에센스/세럼/앰플  메이크프렘   
229       E230

In [5]:
duplicates_sorted

product_id   category brand  \
183       E184  에센스/세럼/앰플    VT   
12        E013  에센스/세럼/앰플    VT   
88        E089  에센스/세럼/앰플    VT   
31        E032  에센스/세럼/앰플    VT   
247       E248  에센스/세럼/앰플    구달   
..         ...        ...   ...   
164       E165  에센스/세럼/앰플  프랭클리   
149       E150  에센스/세럼/앰플  피지오겔   
99        E100  에센스/세럼/앰플  피지오겔   
19        E020  에센스/세럼/앰플  피지오겔   
298       E299  에센스/세럼/앰플  피지오겔   

                                          product_name  \
183  [모공에센스] VT 리들샷 100 에센스 50ml (+리들샷 300 1ml*3ea기...   
12   [모공에센스] VT 리들샷 100 에센스 50ml 기획 (+리들샷300 1ml*3개...   
88                         [모공/흡수] VT 리들샷 300 에센스 50ml   
31   [모공흡수] VT 리들샷 300 에센스 30ml 기획 (+리들샷 300 파우치 1m...   
247         [리뉴얼] 구달 청귤 비타C 잡티케어 세럼 알파 30ml 기획 (+20ml)   
..                                                 ...   
164                                   프랭클리 클로저 세럼 30ml   
149          피지오겔 레드수딩 AI 리페어 에센스 200ml 기획(+리페어크림30ml)   
99    [1월 올영픽] 피지오겔 레드수딩 AI 리페어 에센스 200ml 기획(+로션 50ml)   
19         [1월 올영픽] 피지오겔 사이언수티컬즈 데일리뮨 앰플 세럼 20ml 2입 기획   
298  [비타민C 앰플] 피지오겔 사이언수티컬즈 데일리뮨 앰플 세럼 20ml + 4ml +...   

                                           ingredients  product_rating  \
183  정제수, 다이프로필렌글라이콜, 글리세린, 나이아신아마이드, 부틸렌글라이콜, 마카다미...             4.7   
12   정제수, 다이프로필렌글라이콜, 글리세린, 나이아신아마이드, 부틸렌글라이콜, 마카다미...             4.8   
88   정제수, 다이프로필렌글라이콜, 글리세린, 나이아신아마이드, 부틸렌글라이콜, 마카다미...             4.8   
31   정제수, 다이프로필렌글라이콜, 글리세린, 나이아신아마이드, 부틸렌글라이콜, 마카다미...             4.8   
247  정제수,탄제린추출물,부틸렌글라이콜,나이아신아마이드,다이프로필렌글라이콜,메틸글루세스-...             4.8   
..                                                 ...             ...   
164  정제수, 글리세린, 부틸렌글라이콜, 프로판다이올, 1,2-헥산다이올, 하이드롤라이즈...             4.7   
149  피지오겔 레드수딩 AI 리페어 크림 정제수, 올리브오일, 글리세린, 펜틸렌글라이콜,...             4.9   
99   피지오겔 레드수딩 AI 리페어 크림 정제수, 올리브오일, 글리세린, 펜틸렌글라이콜,...             4.9   
19   정제수, 다이프로필렌글라이콜, 프로판다이올, 아스코빅애씨드, 2,3-부탄다이올, 글...             4.9   
298  정제수, 다이프로필렌글라이콜, 프로판다이올, 아스코빅애씨드, 2,3-부탄다이올, 글...             4.8   

              canonical_name  
183           VT 리들샷 100 에센스  
12            VT 리들샷 100 에센스  
88            VT 리들샷 300 에센스  
31            VT 리들샷 300 에센스  
247     구달 청귤 비타C 잡티케어 세럼 알파  
..                       ...  
164              프랭클리 클로저 세럼  
149     피지오겔 레드수딩 AI 리페어 에센스  
99      피지오겔 레드수딩 AI 리페어 에센스  
19   피지오겔 사이언수티컬즈 데일리뮨 앰플 세럼  
298  피지오겔 사이언수티컬즈 데일리뮨 앰플 세럼  

[94 rows x 7 columns]

In [26]:
# canonical_name이 중복된 행들 중 첫 번째(위쪽) 행만 남기고 나머지 삭제

import pandas as pd

# 원본 데이터프레임 확인
print(f"원본 행 개수: {len(df)}")

# 중복 확인
duplicates = df[df.duplicated(subset=['canonical_name'], keep=False)]
print(f"중복된 행 개수: {len(duplicates)}")

# canonical_name이 중복된 경우, 첫 번째 행만 유지하고 나머지 삭제
df_cleaned = df.drop_duplicates(subset=['canonical_name'], keep='first')

print(f"중복 제거 후 행 개수: {len(df_cleaned)}")
print(f"삭제된 행 개수: {len(df) - len(df_cleaned)}")

# 결과 확인
print("\n중복 제거 후 데이터:")
print(df_cleaned.head())

# 원본 df를 업데이트하려면 (선택사항):
# df = df_cleaned

원본 행 개수: 310
중복된 행 개수: 94
중복 제거 후 행 개수: 259
삭제된 행 개수: 51

중복 제거 후 데이터:
  product_id   category  brand  \
0       E001  에센스/세럼/앰플    아누아   
1       E002  에센스/세럼/앰플    차앤박   
2       E003  에센스/세럼/앰플    메디힐   
3       E004  에센스/세럼/앰플   메디큐브   
4       E005  에센스/세럼/앰플  이니스프리   

                                                        product_name  \
0     [2025 어워즈] 아누아 피디알엔 히알루론산 캡슐 100 세럼 30mL (+리필30mL+크림10mL+패드2매)   
1                    [1월올영픽/진정탄력] 차앤박 더마앤서 액티브 부스트 PDRN 앰플 30ml 더블기획   
2                       [단독/1+1] 메디힐 마데카소사이드 흔적 리페어 세럼 40+40ml 더블 기획   
3  [2025 어워즈/미백천재앰플] 메디큐브 PDRN 핑크앰플 30ml 어워즈 기획 (+리필50ml+세럼1.5ml*5매)   
4                            [1월올영픽/키링증정] 이니스프리 레티놀 시카 앰플 30ml 더블 기획   

                                                                                                                                                                                                                                                                                                   

In [27]:
df_cleaned.to_csv("crawling_data/review_only_100_data/table3_essence_ingredients_productname_deduplicated.csv", index=False)

In [28]:
df_cleaned.head(30)
df_cleaned.to_excel("crawling_data/review_only_100_data/table3_essence_ingredients_productname_deduplicated.xlsx", index=False)

In [29]:
df_cleaned.head()

product_id   category  brand  \
0       E001  에센스/세럼/앰플    아누아   
1       E002  에센스/세럼/앰플    차앤박   
2       E003  에센스/세럼/앰플    메디힐   
3       E004  에센스/세럼/앰플   메디큐브   
4       E005  에센스/세럼/앰플  이니스프리   

                                                        product_name  \
0     [2025 어워즈] 아누아 피디알엔 히알루론산 캡슐 100 세럼 30mL (+리필30mL+크림10mL+패드2매)   
1                    [1월올영픽/진정탄력] 차앤박 더마앤서 액티브 부스트 PDRN 앰플 30ml 더블기획   
2                       [단독/1+1] 메디힐 마데카소사이드 흔적 리페어 세럼 40+40ml 더블 기획   
3  [2025 어워즈/미백천재앰플] 메디큐브 PDRN 핑크앰플 30ml 어워즈 기획 (+리필50ml+세럼1.5ml*5매)   
4                            [1월올영픽/키링증정] 이니스프리 레티놀 시카 앰플 30ml 더블 기획   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     ingredients  \
0  [세럼] 하이드롤라이즈드하이알루로닉애씨드(29,049ppm), 나이아신아마이드, 하이드롤라이즈드콜라겐, 소듐하이알루로 네이트(800ppm), 아데노신, 소듐디엔에이(100ppm), 하이알 루로닉애씨드(30ppm), 하이드롤라이즈드소듐하이알루로네이 트(30ppm), 하이드록시프로필트라이모늄하이알루로네이트 (30ppm), 포타슘하이알루로네이트(30ppm), 소듐하이알루 로네이트크로스폴리머(30ppm), 소듐아세틸레이티드하이알 루로네이트(1ppm), 시트릭애씨드 * 피디알엔: 소듐디엔에이 [크림] 정제수, 스쿠알란, 글리세린, 나이아신아마이드, 펜틸렌글라이콜, 1,2-헥산다이올, 소듐디엔에이, 하이드롤라이즈드콜라겐, 소듐하이알루로네이트, 인도멀구슬나무잎추출물, 하이드롤라이즈드하이알루로닉애씨드, 인도멀구슬나무꽃추출물, 아이비고드열매추출물, 하이알루로닉애씨드, 알로에베라꽃추출물, 가지열매추출물, 홀리바질잎추출물, 참산호말추출물, 울금뿌리추출물, 병풀잎추출물, 다이페닐실록시페닐트라이메티콘, 피씨에이다이메티콘, 부틸렌글라이콜, 프로판다이올, 암모늄아크릴로일다이메틸타우레이트/브이피코폴리머, 다이글리세린, 하이드록시에틸아크릴레이트/소듐아크릴로일다이메틸타우레이트코폴리머, 아크릴레이트/C10-30알킬아크릴레이트크로스폴리머, C12-20알킬글루코사이드, 메틸프로판다이올, 에틸헥실글리세린, 아데노신, 소듐파이테이트, 아미노메틸프로판다이올, 피브이엠/엠에이코폴리머, 마데카소사이드, C12-13알케스-9, 다이메틸실란올하이알루로네이트, 하이드롤라이즈드소듐하이알루로네이트, 아시아티코사이드, 포타슘하이알루로네이트, 하이드록시프로필트라이모늄하이알루로네이트, 소듐하이알루로네이트크로스폴리머, 소듐하이알루로네이트다이메틸실란올, 소듐아세틸레이티드하이알루로네이트, 베헤닐알코올, C14-22알코올, 카보머 [패드] 정제수, 프로판다이올, 메틸프로판다이올, 글리세레스-26, 1,2-헥산다이올, 트레할로오스, 하이드록시아세토페논, 다이에톡시에틸석시네이트, 나이아신아마이드, 부틸렌글라이콜, 하이드록시에틸우레아, 소듐시트레이트, 글리세린, 황련뿌리추출물, 시트릭애씨드, 다이소듐이디티에이, 메틸다이아이소프로필프로피온아마이드, 녹차카테킨, 잔탄검, 판테놀, 카페인, 덱스트린, 카카오씨추출물, 하이드롤라이즈드소듐하이알루로네이트, 제비꽃꽃추출물, 감나무잎추출물, 참마뿌리추출물, 다시마추출물, 알로에베라잎추출물, 포도추출물, 당느릅나무뿌리추출물, 에틸헥실글리세린, 잇꽃꽃추출물, 호장근뿌리추출물, 커피콩추출물, 초피나무열매추출물, 밤껍질추출물, 녹차추출물, 아스코빅애씨드, 소듐하이알루로네이트, 병풀추출물, 탄닉애씨드, 하이드록시프로필트라이모늄하이알루로네이트, 징크옥사이드, 소듐아세틸레이티드하이알루로네이트, 하이드롤라이즈드하이알루로닉애씨드, 사포닌, 트라이에톡시카프릴릴실레인, 하이알루로닉애씨드, 미네랄솔트, 소듐하이알루로네이트크로스폴리머, 포타슘하이알루로네이트   
1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           

In [30]:
review = pd.read_csv("crawling_data/review_only_100_data/table1_essence_reviews.csv")
review.head()

product_id                                                    product_name  \
0       E001  [2025 어워즈] 아누아 피디알엔 히알루론산 캡슐 100 세럼 30mL (+리필30mL+크림10mL+패드2매)   
1       E001  [2025 어워즈] 아누아 피디알엔 히알루론산 캡슐 100 세럼 30mL (+리필30mL+크림10mL+패드2매)   
2       E001  [2025 어워즈] 아누아 피디알엔 히알루론산 캡슐 100 세럼 30mL (+리필30mL+크림10mL+패드2매)   
3       E001  [2025 어워즈] 아누아 피디알엔 히알루론산 캡슐 100 세럼 30mL (+리필30mL+크림10mL+패드2매)   
4       E001  [2025 어워즈] 아누아 피디알엔 히알루론산 캡슐 100 세럼 30mL (+리필30mL+크림10mL+패드2매)   

  user_id          user_keywords  user_rating  \
0     망구네         민감성 | 웜톤 | 트러블            5   
1    하이눈렐    건성 | 여름쿨톤 | 미백 | 탄력            4   
2     소와양     복합성 | 웜톤 | 잡티 | 주름            5   
3      배뽀  복합성 | 봄웜톤 | 잡티 | 블랙헤드            5   
4    귤스귤스           지성 | 웜톤 | 주름            5   

                                                                                                                                                                                                                                                                             review_text  
0                                                                                                                                                                                                                                                        언니 심부름으로 구매한 제품인데 유명하다고 구매해달랫어요  
1                                                                                                             수분세럼으론 좋지만 pdrn 100ppm 들어가있어요 0.01%임 과장광고 개쩜~ 바르는 리쥬란이니 뭐니 말도안되는 소리 그만하시구연 기능성 생각하고 사면 걍 원효대사 해골물 되는거여~ 그냥 수분세럼용으로 매일 쓰는데 쫀쫜한 느낌이 있어서 건성한텐 좋아요~~ 사진은 대용량(50ml)샀다가 리필 채운거에요  
2                                                                                                                                                                                                                                      아노와 히알루론션. 세럼 좋아요. 잘 사용하고 있습니다. 축축해요. 촉촉함이 오래 갑니다  
3  🩵구매이유 캡슐로 된 걸 먼저 구매해서 사용하고 있었는데 마침 세럼이 다 떨어져가서 메이크업용 세럼으로 구매했어요! 🩵보습 한번 바르기만 해도 스파 다녀온 것처럼 얼굴에 광택감이 어마무시하게 올라오는데요 피부속까지 쫀쫀하게 만들어준답니다>< 🩵제형 아주 묽은 제형이라 흘러내리는데 바르고나면 얼굴에 착 밀착이 되면서 살짝 끈적이는 듯 한데 저는 같이 사용하는 크림을 발라주면 조합이 딱 맞더라구요✨ 🩵지금 아누아 피디알엔 토너랑 같이 사용해주고 있는데 이렇게 써주니까 효과가 더 좋은 것 같아영!  
4                                                                                                                                                                                                                                               속건조에 효과가 좋다고 해서 구매했어요~ 겨울철에 너무 좋은제품이예요~~

In [31]:
# df_cleaned의 상품명과 완전히 일치하는 리뷰만 필터링

import pandas as pd

# df_cleaned 읽기
df_cleaned = pd.read_csv('crawling_data/review_only_100_data/table1_essence_products_cleaned.csv')

# 원본 review 데이터 읽기 (필터링 전)
review = pd.read_csv("crawling_data/review_only_100_data/table1_essence_reviews.csv")

print(f"df_cleaned 상품 수: {len(df_cleaned)}")
print(f"원본 review 리뷰 수: {len(review)}")

# df_cleaned의 상품명 목록
valid_product_names = set(df_cleaned['product_name'])
print(f"유효한 상품명 수: {len(valid_product_names)}")

# review에서 상품명이 df_cleaned에 있는 것만 필터링
review_filtered = review[review['product_name'].isin(valid_product_names)].copy()

print(f"\n필터링 후 리뷰 수: {len(review_filtered)}")
print(f"제거된 리뷰 수: {len(review) - len(review_filtered)}")

# product_name을 기준으로 새로운 product_id 매핑
product_name_to_id = dict(zip(df_cleaned['product_name'], df_cleaned['product_id']))

# product_id 업데이트
review_filtered['product_id'] = review_filtered['product_name'].map(product_name_to_id)

# 매핑 확인
unmapped = review_filtered[review_filtered['product_id'].isna()]
if len(unmapped) > 0:
    print(f"\n경고: 매핑되지 않은 리뷰 {len(unmapped)}개")
else:
    print("\n모든 리뷰 매핑 성공!")

# product_id로 정렬 후 인덱스 리셋
review_filtered = review_filtered.sort_values('product_id').reset_index(drop=True)

# 결과 확인
print(f"\nreview_filtered에 있는 고유 상품 수: {review_filtered['product_id'].nunique()}")
print(f"첫 번째 product_id: {review_filtered['product_id'].iloc[0]}")
print(f"마지막 product_id: {review_filtered['product_id'].iloc[-1]}")

# 각 product_id별 리뷰 개수 확인
print("\nproduct_id별 리뷰 개수:")
print(review_filtered['product_id'].value_counts().sort_index())

# CSV 저장
review_filtered.to_csv('crawling_data/review_only_100_data/table1_essence_reviews_filtered.csv', index=False, encoding='utf-8-sig')

print("\n파일 저장 완료!")

df_cleaned 상품 수: 259
원본 review 리뷰 수: 31000
유효한 상품명 수: 259

필터링 후 리뷰 수: 23200
제거된 리뷰 수: 7800

모든 리뷰 매핑 성공!

review_filtered에 있는 고유 상품 수: 232
첫 번째 product_id: E001
마지막 product_id: E258

product_id별 리뷰 개수:
product_id
E001    100
E002    100
E003    100
E004    100
E005    100
E006    100
E007    100
E008    100
E009    100
E010    100
E011    100
E012    100
E013    100
E014    100
E015    100
E016    100
E017    100
E018    100
E019    100
E020    100
E021    100
E022    100
E023    100
E025    100
E026    100
E027    100
E029    100
E030    100
E031    100
E032    100
E033    100
E034    100
E035    100
E036    100
E037    100
E038    100
E039    100
E040    100
E041    100
E042    100
E043    100
E044    100
E045    100
E047    100
E048    100
E049    100
E050    100
E051    100
E052    100
E053    100
E054    100
E055    100
E056    100
E057    100
E058    100
E059    100
E060    100
E061    100
E062    100
E063    100
E064    100
E065    100
E066    100
E067    100
E068    100
E069  

In [33]:
review_filtered['product_id'].nunique()

232

In [34]:
df_cleaned['product_id'].nunique()

259

In [35]:
# df_cleaned의 모든 product_id
all_products = set(df_cleaned['product_id'])

# review_filtered에 있는 product_id
products_with_reviews = set(review_filtered['product_id'])

# 리뷰가 없는 상품들
products_without_reviews = all_products - products_with_reviews

print(f"리뷰가 없는 상품 수: {len(products_without_reviews)}")
print(f"리뷰가 없는 상품 ID:")
print(sorted(products_without_reviews))

# 리뷰가 없는 상품 상세 정보
print("\n리뷰가 없는 상품 정보:")
print(df_cleaned[df_cleaned['product_id'].isin(products_without_reviews)])

리뷰가 없는 상품 수: 27
리뷰가 없는 상품 ID:
['E024', 'E028', 'E046', 'E092', 'E155', 'E233', 'E234', 'E235', 'E236', 'E238', 'E239', 'E240', 'E241', 'E242', 'E243', 'E244', 'E245', 'E246', 'E247', 'E248', 'E249', 'E253', 'E254', 'E255', 'E256', 'E257', 'E259']

리뷰가 없는 상품 정보:
    product_id   category    brand  \
23        E024  에센스/세럼/앰플      리덴스   
27        E028  에센스/세럼/앰플       구달   
45        E046  에센스/세럼/앰플      닥터지   
91        E092  에센스/세럼/앰플      로벡틴   
154       E155  에센스/세럼/앰플    디오디너리   
232       E233  에센스/세럼/앰플     에스트라   
233       E234  에센스/세럼/앰플     마녀공장   
234       E235  에센스/세럼/앰플    라로슈포제   
235       E236  에센스/세럼/앰플    바이오더마   
237       E238  에센스/세럼/앰플     에스테덤   
238       E239  에센스/세럼/앰플    바이오더마   
239       E240  에센스/세럼/앰플      라네즈   
240       E241  에센스/세럼/앰플     아이오페   
241       E242  에센스/세럼/앰플      시드물   
242       E243  에센스/세럼/앰플     라타플랑   
243       E244  에센스/세럼/앰플    닥터자르트   
244       E245  에센스/세럼/앰플     스킨앤랩   
245       E246  에센스/세럼/앰플  더랩바이블랑두   
246       E247  

In [40]:
review[review['product_name']=='대신 이 상품은 어떤가요?']['product_id'].unique()

array(['E269', 'E271', 'E272', 'E273', 'E274', 'E275', 'E276', 'E277',
       'E278', 'E280', 'E281', 'E282', 'E283', 'E284', 'E285', 'E286',
       'E287', 'E288', 'E289', 'E290', 'E291', 'E292', 'E293', 'E294',
       'E295', 'E297', 'E299', 'E300', 'E303', 'E304', 'E305', 'E306',
       'E307', 'E309', 'E310'], dtype=object)

In [42]:
# review 데이터의 product_name을 table3_essence_ingredients.csv의 product_name으로 대체

import pandas as pd

# 데이터 로드
review = pd.read_csv("crawling_data/review_only_100_data/table1_essence_reviews.csv")
table3 = pd.read_csv("crawling_data/review_only_100_data/table3_esssence_ingredients.csv")

print(f"원본 review 리뷰 수: {len(review)}")
print(f"table3 상품 수: {len(table3)}")

# table3에서 product_id와 product_name 매핑 딕셔너리 생성
correct_product_names = dict(zip(table3['product_id'], table3['product_name']))

print(f"\n매핑 딕셔너리 생성: {len(correct_product_names)}개")

# review의 product_name을 올바른 이름으로 대체
review['product_name'] = review['product_id'].map(correct_product_names)

# 매핑되지 않은 항목 확인
unmapped = review[review['product_name'].isna()]
if len(unmapped) > 0:
    print(f"\n경고: 매핑되지 않은 리뷰 {len(unmapped)}개")
    print(f"매핑되지 않은 product_id: {unmapped['product_id'].unique()}")
else:
    print("\n모든 리뷰의 product_name 대체 성공!")

# 결과 확인
print(f"\nreview에 있는 고유 product_id 수: {review['product_id'].nunique()}")
print("\n수정된 review 데이터:")
print(review.head(10))

# CSV 저장
review.to_csv('crawling_data/review_only_100_data/table1_essence_reviews_corrected.csv', index=False, encoding='utf-8-sig')

print("\n파일 저장 완료: table1_essence_reviews_corrected.csv")

원본 review 리뷰 수: 31000
table3 상품 수: 310

매핑 딕셔너리 생성: 310개

모든 리뷰의 product_name 대체 성공!

review에 있는 고유 product_id 수: 310

수정된 review 데이터:
  product_id                                                    product_name  \
0       E001  [2025 어워즈] 아누아 피디알엔 히알루론산 캡슐 100 세럼 30mL (+리필30mL+크림10mL+패드2매)   
1       E001  [2025 어워즈] 아누아 피디알엔 히알루론산 캡슐 100 세럼 30mL (+리필30mL+크림10mL+패드2매)   
2       E001  [2025 어워즈] 아누아 피디알엔 히알루론산 캡슐 100 세럼 30mL (+리필30mL+크림10mL+패드2매)   
3       E001  [2025 어워즈] 아누아 피디알엔 히알루론산 캡슐 100 세럼 30mL (+리필30mL+크림10mL+패드2매)   
4       E001  [2025 어워즈] 아누아 피디알엔 히알루론산 캡슐 100 세럼 30mL (+리필30mL+크림10mL+패드2매)   
5       E001  [2025 어워즈] 아누아 피디알엔 히알루론산 캡슐 100 세럼 30mL (+리필30mL+크림10mL+패드2매)   
6       E001  [2025 어워즈] 아누아 피디알엔 히알루론산 캡슐 100 세럼 30mL (+리필30mL+크림10mL+패드2매)   
7       E001  [2025 어워즈] 아누아 피디알엔 히알루론산 캡슐 100 세럼 30mL (+리필30mL+크림10mL+패드2매)   
8       E001  [2025 어워즈] 아누아 피디알엔 히알루론산 캡슐 100 세럼 30mL (+리필30mL+크림10mL+패드2매)   
9       E001  [2025 어워즈] 아누아 피디알엔 히알루론산 캡슐 100 세럼 30mL (+리필30mL+크

In [43]:
table1_essence_reviews_corrected = pd.read_csv('crawling_data/review_only_100_data/table1_essence_reviews_corrected.csv')
table1_essence_reviews_corrected.head()
table1_essence_reviews_corrected['product_name'].unique()

array(['[2025 어워즈] 아누아 피디알엔 히알루론산 캡슐 100 세럼 30mL (+리필30mL+크림10mL+패드2매)',
       '[1월올영픽/진정탄력] 차앤박 더마앤서 액티브 부스트 PDRN 앰플 30ml 더블기획',
       '[단독/1+1] 메디힐 마데카소사이드 흔적 리페어 세럼 40+40ml 더블 기획',
       '[2025 어워즈/미백천재앰플] 메디큐브 PDRN 핑크앰플 30ml 어워즈 기획 (+리필50ml+세럼1.5ml*5매)',
       '[1월올영픽/키링증정] 이니스프리 레티놀 시카 앰플 30ml 더블 기획',
       '[2025 어워즈 1위] 토리든 다이브인 저분자 히알루론산 세럼 100ml 어워즈 한정기획',
       '[1월올영픽/보습광채] 차앤박 프로폴리스 에너지 액티브 앰플 30ml 2입 기획 (+15ml)',
       '[2025 어워즈] 웰라쥬 리얼 히알루로닉 블루 100 앰플 75ml 더블 어워즈 한정기획',
       '[1월 올영픽 한정기획] 아이오페 레티놀 슈퍼 바운스 세럼 30ml +25ml 기획',
       '[1월 올영픽 한정기획] 아이오페 레티놀 레티젝션 세럼 30ml +25ml 기획',
       '[1위 속보습세럼/단독기획] 토리든 다이브인 저분자 히알루론산 세럼 50ml 리필기획(+리필팩 50ml)',
       '[1월올영픽]성분에디터 실크펩타이드 EGF 하트핏 볼륨 리프팅 앰플 40ml 더블기획(+산리오캐릭터즈 파우치)',
       '[모공에센스] VT 리들샷 100 에센스 50ml 기획 (+리들샷300 1ml*3개+PDRN에센스 1.5ml*3ea)',
       '[애니PICK][광채탄력] VT PDRN 에센스 30ml 기획 2종 (택 1)',
       '[1월올영PICK] 브링그린 징크테카 트러블 세럼 (대용량/콜라보/기획)',
       '[트러블1등/단독기획] 파티온 노스카나인 트러블 세럼 50ml 기획(+15ml+마스크 1매)',
       

In [44]:
# df_cleaned의 상품명과 완전히 일치하는 리뷰만 필터링

import pandas as pd

# df_cleaned 읽기
df_cleaned = pd.read_csv('crawling_data/review_only_100_data/table1_essence_products_cleaned.csv')

# 원본 review 데이터 읽기 (필터링 전)
review = pd.read_csv("crawling_data/review_only_100_data/table1_essence_reviews_corrected.csv")

print(f"df_cleaned 상품 수: {len(df_cleaned)}")
print(f"원본 review 리뷰 수: {len(review)}")

# df_cleaned의 상품명 목록
valid_product_names = set(df_cleaned['product_name'])
print(f"유효한 상품명 수: {len(valid_product_names)}")

# review에서 상품명이 df_cleaned에 있는 것만 필터링
review_filtered = review[review['product_name'].isin(valid_product_names)].copy()

print(f"\n필터링 후 리뷰 수: {len(review_filtered)}")
print(f"제거된 리뷰 수: {len(review) - len(review_filtered)}")

# product_name을 기준으로 새로운 product_id 매핑
product_name_to_id = dict(zip(df_cleaned['product_name'], df_cleaned['product_id']))

# product_id 업데이트
review_filtered['product_id'] = review_filtered['product_name'].map(product_name_to_id)

# 매핑 확인
unmapped = review_filtered[review_filtered['product_id'].isna()]
if len(unmapped) > 0:
    print(f"\n경고: 매핑되지 않은 리뷰 {len(unmapped)}개")
else:
    print("\n모든 리뷰 매핑 성공!")

# product_id로 정렬 후 인덱스 리셋
review_filtered = review_filtered.sort_values('product_id').reset_index(drop=True)

# 결과 확인
print(f"\nreview_filtered에 있는 고유 상품 수: {review_filtered['product_id'].nunique()}")
print(f"첫 번째 product_id: {review_filtered['product_id'].iloc[0]}")
print(f"마지막 product_id: {review_filtered['product_id'].iloc[-1]}")

# 각 product_id별 리뷰 개수 확인
print("\nproduct_id별 리뷰 개수:")
print(review_filtered['product_id'].value_counts().sort_index())

# CSV 저장
review_filtered.to_csv('crawling_data/review_only_100_data/table1_essence_reviews_filtered.csv', index=False, encoding='utf-8-sig')

print("\n파일 저장 완료!")

df_cleaned 상품 수: 259
원본 review 리뷰 수: 31000
유효한 상품명 수: 259

필터링 후 리뷰 수: 25900
제거된 리뷰 수: 5100

모든 리뷰 매핑 성공!

review_filtered에 있는 고유 상품 수: 259
첫 번째 product_id: E001
마지막 product_id: E259

product_id별 리뷰 개수:
product_id
E001    100
E002    100
E003    100
E004    100
E005    100
E006    100
E007    100
E008    100
E009    100
E010    100
E011    100
E012    100
E013    100
E014    100
E015    100
E016    100
E017    100
E018    100
E019    100
E020    100
E021    100
E022    100
E023    100
E024    100
E025    100
E026    100
E027    100
E028    100
E029    100
E030    100
E031    100
E032    100
E033    100
E034    100
E035    100
E036    100
E037    100
E038    100
E039    100
E040    100
E041    100
E042    100
E043    100
E044    100
E045    100
E046    100
E047    100
E048    100
E049    100
E050    100
E051    100
E052    100
E053    100
E054    100
E055    100
E056    100
E057    100
E058    100
E059    100
E060    100
E061    100
E062    100
E063    100
E064    100
E065    100
E066  

최종 259개 상품만 남긴 성분과 리뷰 데이터셋 
- table1_essence_reviews_filtered
-  table1_essence_products_cleaned